packages/components/modules/profiles/ProfilePopover/AddProfileModal/index.tsx

```diff
@@ -0,0 +1,212 @@
+import { FC } from 'react'
+
+import { TextField } from '@baseapp-frontend/design-system'
+import { setFormRelayErrors } from '@baseapp-frontend/utils'
+
+import { zodResolver } from '@hookform/resolvers/zod'
+import { LoadingButton } from '@mui/lab'
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Link,
+  Typography,
+} from '@mui/material'
+import { useForm } from 'react-hook-form'
+import { ConnectionHandler } from 'react-relay'
+import slugify from 'slugify'
+import z from 'zod'
+
+import { Form } from '../../../__shared__/SocialInput/styled'
+import { useOrganizationCreateMutation } from '../../graphql/mutations/OrganizationCreate'
+
+const slugRegex = /^[a-z0-9]+(?:-[a-z0-9]+)*$/
+
+const schema = z.object({
+  name: z
+    .string()
+    .trim()
+    .min(3, { message: 'Must have at least three character' })
+    .max(255, { message: 'Must have at most 255 characters' }),
+  urlPath: z
+    .string()
+    .regex(slugRegex, 'Must contain only lowercase letters, numbers, and hyphens')
+    .max(500, { message: 'Must have at most 500 characters' }),
+})
+
+let nextClientMutationId = 0
+
+export interface AddProfileModalProps {
+  addNewProfileLabel?: string
+  termsAndConditionsUrl?: string
+  addNewProfileDescription?: string
+  onClose?: () => void
+  open: boolean
+  setOpen: (open: boolean) => void
+  submitLabel?: string
+}
+
+interface OrganizationCreateForm {
+  name: string
+  urlPath: string
+}
+
+const AddProfileModal: FC<AddProfileModalProps> = ({
+  addNewProfileLabel = 'New organization',
+  termsAndConditionsUrl = '',
+  addNewProfileDescription = 'Create an organization and invite multiple members to manage and collaborate.',
+  submitLabel = 'Create Organization',
+  onClose,
+  open,
+  setOpen,
+}) => {
+  const form = useForm<OrganizationCreateForm>({
+    defaultValues: {
+      name: '',
+      urlPath: '',
+    },
+    resolver: zodResolver(schema),
+  })
+
+  const [commitMutation, isMutationInFlight] = useOrganizationCreateMutation()
+
+  const handleClose = () => {
+    setOpen(false)
+    form.reset()
+    onClose?.()
+  }
+
+  const onSubmit = (data: OrganizationCreateForm) => {
+    if (isMutationInFlight) return
+
+    nextClientMutationId += 1
+    const clientMutationId = nextClientMutationId.toString()
+
+    commitMutation({
+      variables: {
+        input: {
+          name: data.name,
+          urlPath: data.urlPath,
+          clientMutationId,
+        },
+      },
+      updater: (store) => {
```

Comment:
we probably don't need an `updater` function for this case.  We could try to use `@appendEdge` or `@prependEdge` in order to add the new edge to the list, so on the mutation query it would be:

```
export const OrganizationCreateMutationQuery = graphql`
  mutation OrganizationCreateMutation($input: OrganizationCreateInput!, $connections: [ID!]!) {
    organizationCreate(input: $input) {
      organization @prependEdge(connections: $connections) {
        node {
          id
        }
      }
      profile {
        node {
          ...ProfileItemFragment
        }
      }
      errors {
        field
        messages
      }
    }
  }
`
```

but for that we need to make sure we have a `connection` on the query (which it already have `@connection(key: "ProfilesListFragment_profiles") `). SO know we just need to make sure we passt the connection id on the commit function, something like:
```
   const connectionID = ConnectionHandler.getConnectionID(roomId, 'ProfilesListFragment_profiles')
    
    commitMutation({
      variables: {
        input: {
          name: data.name,
          urlPath: data.urlPath,
          clientMutationId,
        },
        connections: [connectionID],
      }
```

[adding edges docs](https://relay.dev/docs/guided-tour/list-data/updating-connections/#using-declarative-directives)
[connections docs](https://relay.dev/docs/guided-tour/list-data/connections/)

Reply:
@anicioalexandre exactly! just point that the `connectionId` should be `const connectionID = ConnectionHandler.getConnectionID(user.id, 'ProfilesListFragment_profiles')`. `user.id` instead of `roomId`

Reply:
Updated @nossila and @anicioalexandre